### PR TITLE
ADUCM3029 platform set irq priority

### PR DESCRIPTION
**Pull request by rbolboac**
- Added implementation for IRQ set_priority API for aducm3029 platform for NVIC controller.
- Added implementation for IRQ set_priority API for aducm3029 platform for GPIO controller.
- Added extra checks for irq_id in case of external interrupt event for GPIO controller.
- Fixed the trigger level setting for GPIO controller.
- Added GPIO trigger example for aducm3029 platform for adxrs290-pmdz project.

### CODE REVIEW

File: drivers/platform/aducm3029/aducm3029_gpio_irq.c [hunk 1]
```diff
@@ -475,6 +497,52 @@ static int aducm_gpio_irq_global_disable(struct no_os_irq_ctrl_desc *desc)
...
+	case ADUCM_GPIO_B_GROUP_SOFT_CTRL:
+		NVIC_SetPriority(ADI_GPIO_INTB_IRQ, priority_level);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
```
Comment (CiprianRegus) [inline drivers/platform/aducm3029/aducm3029_gpio_irq.c:544]: Extra line

File: projects/adxrs290-pmdz/src/platform/aducm3029/parameters.h [hunk 2]
```diff
@@ -60,7 +61,7 @@
 #define UART_BAUDRATE	115200
 
 #define SPI_DEVICE_ID   1
-#define SPI_BAUDRATE    1000000
+#define SPI_BAUDRATE    4000000
```
Comment (CiprianRegus) [inline projects/adxrs290-pmdz/src/platform/aducm3029/parameters.h:64]: Is this done on purpose?

**Review by CiprianRegus — COMMENTED** [review]
Also, the title for the second commit seems to be too long.

File: projects/adxrs290-pmdz/src/platform/aducm3029/parameters.h (hunk 2 above)
Comment (rbolboac) [reply projects/adxrs290-pmdz/src/platform/aducm3029/parameters.h:64]: yes, increasing the SPI baud-rate will allow the whole interrupt-based system to work as expected

Comment (CiprianRegus) [reply projects/adxrs290-pmdz/src/platform/aducm3029/parameters.h:64]: ok...

**Review by CiprianRegus — APPROVED** [review]

**Pull request merged**